File: mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height;
    
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);
    
    for (int i = 1; i <= height; i++)
    {
        for (int j = height; j >= 1; j--)
        {
            if (j > i)
            {
                printf(" ");
            }
            else
            {
                printf("#");
            }
        }
        printf("  ");
        for (int j = 1; j <= height; j++)
        {
            if (j <= i)
            {
                printf("#");
            }
        }
        printf("\n");
    }
}

File: Problem Set 6/mario/more/mario.py
from cs50 import get_int

while True:
    height = get_int("Height: ")
    
    # Break only when the user provides a valid height
    if height > 0 and height < 9:
        break

for i in range(1, height + 1):
    # Number of hashes needed per line, per side
    hashes = "#" * i
    
    # Number of spaces needed before the left side hashes
    spaces = " " * (height - i)
    
    # Nicely formatted print statement
    print("{s}{h}  {h}".format(h=hashes, s=spaces))